build.gradle.kts

```diff
@@ -0,0 +1,559 @@
+            }
+
+            withJavadocJar()
+            withSourcesJar()
+        }
+
+        configure<CheckstyleExtension> {
+            configDirectory.set(file("$rootDir/buildscripts/"))
+            toolVersion = "8.12"
+            isIgnoreFailures = false
+            configProperties["rootDir"] = rootDir
+        }
+
+        configure<JacocoPluginExtension> {
+            toolVersion = "0.8.6"
+        }
+
+        val javaToolchains = the<JavaToolchainService>()
+
+        tasks {
+            withType(Javadoc::class) {
+                exclude("io/opentelemetry/internal/**")
+            }
+
+            val testJava8 by registering(Test::class) {
+                javaLauncher.set(javaToolchains.launcherFor {
+                    languageVersion.set(JavaLanguageVersion.of(14))
+                })
+
+                configure<JacocoTaskExtension> {
+                    enabled = false
+                }
+            }
+
+            val testAdditionalJavaVersions: String? by rootProject
+            if (testAdditionalJavaVersions == "true") {
+                named("check") {
+                    dependsOn(testJava8)
+                }
+            }
+
+            withType(JavaCompile::class) {
+                options.release.set(8)
+
+                options.compilerArgs.addAll(listOf(
+                        "-Xlint:all",
+                        // We suppress the "try" warning because it disallows managing an auto-closeable with
+                        // try-with-resources without referencing the auto-closeable within the try block.
+                        "-Xlint:-try",
+                        // We suppress the "processing" warning as suggested in
+                        // https://groups.google.com/forum/#!topic/bazel-discuss/_R3A9TJSoPM
+                        "-Xlint:-processing",
+                        // We suppress the "options" warning because it prevents compilation on modern JDKs
+                        "-Xlint:-options",
+
+                        // Fail build on any warning
+                        "-Werror"
+                ))
+
+                options.encoding = "UTF-8"
+
+                if (name.contains("Test")) {
+                    // serialVersionUID is basically guaranteed to be useless in tests
+                    options.compilerArgs.add("-Xlint:-serial")
+                }
+
+                (options as ExtensionAware).extensions.configure<ErrorProneOptions> {
+                    disableWarningsInGeneratedCode.set(true)
+                    allDisabledChecksAsWarnings.set(true)
+
+                    // Doesn't currently use Var annotations.
+                    disable("Var") // "-Xep:Var:OFF"
+
+                    // ImmutableRefactoring suggests using com.google.errorprone.annotations.Immutable,
+                    // but currently uses javax.annotation.concurrent.Immutable
+                    disable("ImmutableRefactoring") // "-Xep:ImmutableRefactoring:OFF"
+
+
```

Comment:
there are a couple of extra CR/LFs scattered in here. on purpose, or can we clean this up?